components/server/src/server.rs

```diff
@@ -605,8 +605,21 @@ impl<ER: RaftEngine> TiKVServer<ER> {
 
         let server_config = Arc::new(self.config.server.clone());
 
+        let raft_store = Arc::new(VersionTrack::new(self.config.raft_store.clone()));
+        let mut node = Node::new(
+            self.system.take().unwrap(),
+            &server_config,
+            raft_store.clone(),
+            self.pd_client.clone(),
+            self.state.clone(),
+            self.background_worker.clone(),
+        );
+        node.bootstrap_store_id(engines.engines.clone())
+            .unwrap_or_else(|e| fatal!("failed to bootstrap node id: {}", e));
+
         // Create server
         let server = Server::new(
+            node.id(),
```

Comment:
How about assigning it after creation to avoid another function parameter?

Reply:
it seems hard to assign id after creation.